skimage/feature/_canny.py

```diff
@@ -161,137 +296,51 @@ def canny(image, sigma=1., low_threshold=None, high_threshold=None, mask=None,
         low_threshold = 0.1
     elif use_quantiles:
         if not(0.0 <= low_threshold <= 1.0):
-            raise ValueError("Quantile thresholds must be between 0 and 1.")
+            raise ValueError("quantile thresholds must be between 0 and 1.")
     else:
-        low_threshold = low_threshold / dtype_max
+        low_threshold /= dtype_max
 
     if high_threshold is None:
         high_threshold = 0.2
     elif use_quantiles:
         if not(0.0 <= high_threshold <= 1.0):
-            raise ValueError("Quantile thresholds must be between 0 and 1.")
+            raise ValueError("quantile thresholds must be between 0 and 1.")
     else:
-        high_threshold = high_threshold / dtype_max
+        high_threshold /= dtype_max
 
-    if mask is None:
-        mask = np.ones(image.shape, dtype=bool)
+    if high_threshold < low_threshold:
+        raise ValueError("low_threshold should be lower then high_threshold")
 
-    def fsmooth(x):
-        return img_as_float(gaussian(x, sigma, mode='constant'))
+    # Image filtering
+    smoothed, eroded_mask = _preprocess(image, mask, sigma, 'constant')
```

Comment:
Is there a plan to put the mode in the API? If not, why this is part of the _preprocess function API if this mode is expected for a canny filter?

Reply:
I would love to see it in the API since it was implicit in the previous implementation (see https://github.com/scikit-image/scikit-image/pull/4342#discussion_r354774473)

Reply:
@sciunto, should we expose `mode` in `canny` API?